xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -97,11 +97,19 @@ void CGUIWindowPVRGuide::StopRefreshTimelineItemsThread()
 
 void CGUIWindowPVRGuide::RegisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
   g_EpgContainer.RegisterObserver(this);
+  if (g_PVRTimers)
+    g_PVRTimers->RegisterObserver(this);
+  CGUIWindowPVRBase::RegisterObservers();
 }
 
 void CGUIWindowPVRGuide::UnregisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
+  CGUIWindowPVRBase::UnregisterObservers();
+  if (g_PVRTimers)
```

Comment:
Just to confirm that we are on the same page: problem here is that g_PVRTimers (which is a CPVRTimers *) can get invalid between call to <code>if (g_PVR_Timers)</code> and <code>g_VRTimers->...</code>, right? OMG, we have this pattern all over the place...

> Why don't you use window messages?

 I'm not the original author of this code, but valid point.

 Would it be okay to use window messages inside the PVR core classes, like CPVRTimers? I think so.